core/src/main/java/com/orange/ouds/core/component/InteractionValuesIndication.kt

```diff
@@ -0,0 +1,361 @@
+    private class AnimatableInteractionValue<T, S>(val interactionValue: InteractionValue<T, S>, val animatable: Animatable<S, *>)
+
+    private companion object {
+        val Easing: Easing = CubicBezierEasing(0.3f, 0.3f, 0.2f, 1.0f)
+    }
+
+    private var pressedAnimations: Job? = null
+
+    private var restingAnimations: Job? = null
+
+    private var isPressing = false
+
+    private val animatableInteractionValues: List<AnimatableInteractionValue<*, *>> = interactionValues.map { createAnimatableInteractionValue(it) }
+
+    override fun onAttach() {
+        coroutineScope.launch {
+            interactionSource.interactions.collectLatest { interaction ->
+                isPressing = interaction is PressInteraction.Press
+                when (interaction) {
+                    is PressInteraction.Press -> animateToPressed()
+                    is PressInteraction.Release, is PressInteraction.Cancel -> animateToResting()
+                }
+            }
+        }
+    }
+
+    override fun ContentDrawScope.draw() {
+        this@draw.drawContent()
+        animatableInteractionValues.forEach { it.updateAnimatingPressed() }
+    }
+
+    private fun <T, S> createAnimatableInteractionValue(interactionValue: InteractionValue<T, S>): AnimatableInteractionValue<T, S> {
+        val resting = with(interactionValue) { toAnimatableValue(resting) }
+
+        @Suppress("UNCHECKED_CAST")
+        val animatable = when (resting) {
+            is Color -> androidx.compose.animation.Animatable(resting)
+            is Float -> Animatable(resting)
+            else -> error("toAnimatableValue must return either a Compose Color or a Float.")
+        } as Animatable<S, *>
+
+        return AnimatableInteractionValue(interactionValue, animatable)
+    }
+
+    private fun animateToPressed() {
+        // Finish any existing animations, in case of a new press while we are still showing animations for a previous one
+        restingAnimations?.cancel()
+        pressedAnimations?.cancel()
+        pressedAnimations = coroutineScope.launch {
+            // Launch all pressed animations at once
+            animatableInteractionValues.map { async { animateToPressed(it) } }.awaitAll()
+        }
+    }
+
+    private fun animateToResting() {
+        restingAnimations = coroutineScope.launch {
+            // Wait for the existing press animations to finish if they are still ongoing
+            pressedAnimations?.join()
+            // Launch all resting animations at once
+            animatableInteractionValues.map { async { animateToResting(it) } }.awaitAll()
+        }
+    }
+
+    private suspend fun <T, S> animateToPressed(animatableInteractionValue: AnimatableInteractionValue<T, S>) {
+        with(animatableInteractionValue) {
+            animatable.snapTo(with(interactionValue) { toAnimatableValue(resting) })
+            animatable.animateTo(with(interactionValue) { toAnimatableValue(pressed) }, tween(easing = Easing))
```

Comment:
As discussed together, we can test with the animation spec used for the Switch component.